pkg/controller/instance/instance_controller.go

```diff
@@ -120,31 +120,29 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 					planName = "upgrade"
 				}
 			} else if !reflect.DeepEqual(old.Spec, new.Spec) {
-				for k, v := range new.Spec.Parameters {
-					if old.Spec.Parameters[k] != v {
-						// Find the right parameter in the FV
-						for _, param := range fv.Spec.Parameters {
-							if param.Name == k {
-								planName = param.Trigger
-								ok = true
-							}
+				for k := range parameterDifference(old.Spec.Parameters, new.Spec.Parameters) {
```

Comment:
Will this run 2 `PlanExecutions` if we change 2 different parameters?n . I think that might be problematic.

Reply:
Yep, this is what I wanted to discuss in this PR. Keep in mind that this part of the logic hasn't changed so we have this problem with current _master_. Actually, there are two questions:

1. If the user **removes** a parameter, do we still trigger the original `param.Trigger` plan? Do we trigger anything?
2. With multiple changes, *multiple plans* might be triggered - do we take first/last/random or do we forbid such changes?

Thoughts?

Reply:
I think always one plan is triggered, but which one is it is decided by the last parameter change :/

Reply:
I need to think about this, I'll weigh in today.

Reply:
> Keep in mind that this part of the logic hasn't changed

You're right, currently the value of `planName` gets used later, sothe "last" element that gets changed is the one that runs.

Reply:
> If the user removes a parameter, do we still trigger the original param.Trigger plan? Do we trigger anything?
I think we would run the plan associated to the parameter taht was "changed" e.g. present -> not present, unless the default value for the parameter was the same as the value that existed

Reply:
We probably don't need to fix this here though. Would be good to have issue tracking this though. I think the current thinking is to add validation that will disallow changing parameters that would end up triggering different plans

Reply:
Agreed. We separate this out from this PR.